### PR TITLE
analyzer: Continue if a Maven project could partially be built

### PR DESCRIPTION
**Pull request by mnonnenmacher**
If a ProjectBuildingException occurs during building a project often the
project was partially built and is contained in the exception. Try to
continue in this case because the problems are often caused by build
extensions which might not be required for the dependency analysis.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/618)

### CODE REVIEW

**Comment by sschuberth** [comment]
Commit message nit: Instead of "occurs during building" I'd say "occurs while building" or "occurs during the build of".

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 1]
```diff
@@ -323,7 +324,26 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
             val pomArtifact = mavenRepositorySystem
                     .createArtifact(it.groupId, it.artifactId, it.version, "", "pom")
 
-            projectBuilder.build(pomArtifact, projectBuildingRequest).project
+            try {
+                projectBuilder.build(pomArtifact, projectBuildingRequest).project
+            } catch (e: ProjectBuildingException) {
+                e.showStackTrace()
+
+                val id = "${it.groupId}:${it.artifactId}:${it.version}"
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/MavenSupport.kt]: Can we use our `Artifact.identifier()` extension function here?

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 2]
```diff
@@ -323,7 +324,26 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
...
+
+                val failedProject = e.results.find { projectBuildingResult ->
+                    projectBuildingResult.projectId == id
+                }
+
+                if (failedProject != null) {
+                    log.warn { "There was an error building the project '$id', continuing with the incompletely " +
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/MavenSupport.kt]: Maybe drop "the" in "the project"?

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 3]
```diff
@@ -323,7 +324,26 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
...
+                            "built project: ${e.collectMessages()}" }
+                    failedProject.project
+                } else {
+                    log.error { "Could not build project '$id': ${e.collectMessages()}" }
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/MavenSupport.kt]: Maybe "Failed to build..."?

File: analyzer/src/main/kotlin/MavenSupport.kt (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/MavenSupport.kt]: Done.

File: analyzer/src/main/kotlin/MavenSupport.kt (hunk 2 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/MavenSupport.kt]: Done.

File: analyzer/src/main/kotlin/MavenSupport.kt (hunk 3 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/MavenSupport.kt]: Done.

**Pull request merged**